src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -664,26 +665,34 @@ public String toVariableNameString(final Inventory i) {
 								"The first requires that the player is online and also accepts only part of the name, " +
 								"while the latter doesn't require that the player is online, but the player's name has to be entered exactly.")
 				.usage("")
-				.examples("")
+				.examples("set {_p} to \"Notch\" parsed as a player # returns <none> unless Notch is actually online")
 				.since("1.0")
 				.defaultExpression(new EventValueExpression<>(Player.class))
 				.after("string", "world")
 				.parser(new Parser<Player>() {
 					@Override
 					@Nullable
-					public Player parse(String s, ParseContext context) {
+					public Player parse(String string, ParseContext context) {
 						if (context == ParseContext.COMMAND) {
-							if (s.isEmpty())
+							if (string.isEmpty())
 								return null;
-							if (UUID_PATTERN.matcher(s).matches())
-								return Bukkit.getPlayer(UUID.fromString(s));
-							List<Player> ps = Bukkit.matchPlayer(s);
-							if (ps.size() == 1)
-								return ps.get(0);
-							if (ps.size() == 0)
-								Skript.error(String.format(Language.get("commands.no player starts with"), s));
+							if (UUID_PATTERN.matcher(string).matches())
+								return Bukkit.getPlayer(UUID.fromString(string));
+							String name = string.toLowerCase(Locale.ENGLISH);
+							List<Player> players = new ArrayList<>();
+							for (Player player : Bukkit.getOnlinePlayers()) {
+								String lowerName = player.getName().toLowerCase(Locale.ENGLISH);
+								if (lowerName.equals(name))
+									return player;
+								if (lowerName.startsWith(name))
+									players.add(player);
```

Comment:
It's probably not much of a difference, but wouldn't using a length check in the `startsWith` if be more efficient? One less loop over the string.